code/game/area/areas.dm

```diff
@@ -564,14 +564,16 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 /**
  * Call back when an atom enters an area
  *
- * Sends signals COMSIG_AREA_ENTERED and COMSIG_ENTER_AREA (to the atom)
+ * Sends signals COMSIG_AREA_ENTERED and COMSIG_ENTER_AREA (to a list of atoms)
  *
  * If the area has ambience, then it plays some ambience music to the ambience channel
  */
 /area/Entered(atom/movable/M)
 	set waitfor = FALSE
 	SEND_SIGNAL(src, COMSIG_AREA_ENTERED, M)
-	SEND_SIGNAL(M, COMSIG_ENTER_AREA, src) //The atom that enters the area
+	for(var/AM in M.area_sensitive_contents)
+		var/atom/movable/recipient = AM
```

Comment:
```suggestion
	for(var/atom/movable/recipient as anything in M.area_sensitive_contents)
```

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists